### PR TITLE
GUILib code cleanup

### PR DESCRIPTION
**Pull request by MaxKellermann**

### CODE REVIEW

**Comment by MaxKellermann** [comment]
Unit test crashes due to https://github.com/xbmc/xbmc/pull/9389 and not due to this PR.

**Comment by MaxKellermann** [comment]
Rebased after commit f711253

File: xbmc/guilib/GUIBaseContainer.cpp [hunk 1]
```diff
@@ -1200,11 +1200,11 @@ bool CGUIBaseContainer::GetCondition(int condition, int data) const
 void CGUIBaseContainer::GetCurrentLayouts()
 {
   m_layout = NULL;
-  for (unsigned int i = 0; i < m_layouts.size(); i++)
+  for (auto &i : m_layouts)
```
Comment (garbear) [inline xbmc/guilib/GUIBaseContainer.cpp]: i'd prefer renaming i to "layout" for code clarity, but no biggie

File: xbmc/guilib/GUIControlGroup.cpp [hunk 2]
```diff
@@ -408,23 +383,17 @@ void CGUIControlGroup::UnfocusFromPoint(const CPoint &point)
...
   CGUIControl::UnfocusFromPoint(point);
 }
 
 bool CGUIControlGroup::HasID(int id) const
 {
   if (CGUIControl::HasID(id)) return true;
-  for (ciControls it = m_children.begin(); it != m_children.end(); ++it)
-  {
-    CGUIControl *child = *it;
+  for (auto *child : m_children)
```
Comment (garbear) [inline xbmc/guilib/GUIControlGroup.cpp]: in multi-line fors (even if its just a single if statement) I prefer to leave the {}, but no biggie

**Comment by garbear** [comment]
thanks, I even see some changes that I could apply to my current working code

**Comment by MaxKellermann** [comment]
Rebased & renamed loop variables & restored curly braces.

**Comment by MaxKellermann** [comment]
Travis-CI fails because an include is missing in ServiceManager.h; not related to this PR.

**Comment by MaxKellermann** [comment]
See https://github.com/xbmc/xbmc/pull/9425 for Travis-CI failure fix

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
looks good imo, objections?

File: xbmc/guilib/GUIControlGroup.cpp [hunk 3]
```diff
@@ -350,9 +335,8 @@ bool CGUIControlGroup::IsAnimating(ANIMATION_TYPE animType)
 
   if (IsVisible())
   {
-    for (iControls it = m_children.begin(); it != m_children.end(); ++it)
-    {
-      if ((*it)->IsAnimating(animType))
+    for (auto *control : m_children) {
```
Comment (Paxxi) [inline xbmc/guilib/GUIControlGroup.cpp]: Brace should be on new row

**Comment by Paxxi** [comment]
Formatting needs fixing but after that, no objections

**Comment by MaxKellermann** [comment]
Moved curly braces.

**Pull request merged**